tools/bin/ci_credentials.sh

```diff
@@ -1,26 +1,150 @@
 #!/usr/bin/env bash
 
 . tools/lib/lib.sh
+. tools/lib/gcp-token.sh
 
 set -e
 
+COMMAND_NAME=$1
+
+# all secrets will be loaded if the second argument is not present
+CONNECTOR_FULLNAME=${2:-all}
+CONNECTOR_NAME=`echo ${CONNECTOR_FULLNAME} | rev | cut -d'/' -f1 | rev`
+
+GSM_SCOPES="https://www.googleapis.com/auth/cloud-platform"
+
+declare -A SECRET_MAP
+
+
 function write_standard_creds() {
   local connector_name=$1
   local creds=$2
   local cred_filename=${3:-config.json}
+  local source_name=${4:-github}
 
   [ -z "$connector_name" ] && error "Empty connector name"
-  [ -z "$creds" ] && error "Creds not set for $connector_name"
+  [ -z "$creds" ] && error "!!!!!Creds not set for $connector_name"
+
+  if [[ $CONNECTOR_NAME != "all" && ${connector_name} != ${CONNECTOR_NAME} ]]; then
+    return 0
+  fi
+  local key="${connector_name}#${cred_filename}"
+  [[ -z "${creds}" ]] && error "Don't find data for the connector '${key})"
+  
+  if [ -v SECRET_MAP[${key}] ]; then
+    echo "The connector '${key}' was added before"
+    return 0
+  fi
+  echo "register the secret ${key} from ${source_name}"
+  SECRET_MAP[${key}]="${creds}"
+  return 0
+}
+
+function _write_standard_creds() {
```

Comment:
we can probably lose the `standard` terminology and rename to `write_secret_to_disk`

Reply:
Done